### DISCUSSION TITLE
Download file from storage: TypeError: data.arrayBuffer is not a function

### DISCUSSION THREAD

**Discussion by moacsjr** [supabase/supabase#15146] [Questions]
Hello guys!
I'm using supabase in an Expo/ReactNative app.
When I tried to download a file from a bucket I'm getting this error  "TypeError: data.arrayBuffer is not a function" 

This is my code, and it was generated by chatgpt ;-)

```javascript
async getFileContent(bucketName, fileName) {
    try {
      const { data, error } = await this.supabase.storage
        .from(bucketName)
        .download(fileName);

      if (error) {
        throw new Error(error.message);
      }

      // Must return the content as an ArrayBuffer
      return data.arrayBuffer();
    } catch (error) {
      console.log('Error:', error);  // TypeError: data.arrayBuffer is not a function
    }
  }

```

I' have inspected the value of the 'data' variable and the result of 'typeof' expression returned 'object' and it was supposed to be a 'Blob' type.

Any help is appreciated!

Thanks in advance!

**Comment by GaryAustin1** [comment]
https://supabase.com/docs/guides/getting-started/tutorials/with-expo#create-an-upload-widget
You might check this Expo example from Supabase (has download in it).   In general chatGPT does not do well with Supabase specific code as it was trained before most of this code was out.
